File: Java/J2EE/Practice2/src/main/java/org/sumdu/Registry.java
package org.sumdu;

import org.sumdu.entities.Employee;
import org.sumdu.exceptions.EmployeeInRegistryException;

import java.util.LinkedList;
import java.util.List;

// Це синглтон
public class Registry {

    private static final Registry WorkersRegistry = new Registry();

    private final List<Employee> workers = new LinkedList<>();

    private Registry() {}

    public static Registry getRegistry() {
        return WorkersRegistry;
    }

    public void addNewEmployee(Employee employee) throws EmployeeInRegistryException {
        if (workers.contains(employee)) {
            throw new EmployeeInRegistryException(
                    "Не можна два рази найняти одного й того самого працівника!"
            );
        } else {
            workers.add(employee);
        }
    }

    public void printList() {
        System.out.println("-----------------------------------------");
        System.out.println("Вивожу список зареестрованих працівників:");
        for (Employee element: workers) {
            element.printInfo();
            System.out.println();
        }
        System.out.println("-----------------------------------------");
    }
}

File: C#/J2EE_C#_Version/Practice2/Registry.cs
using Business.Entities;
using Business.Exceptions;

namespace Business {
    public class Registry {
        private static readonly Registry WorkersRegistry = new ();

        private readonly LinkedList<Employee> workers;

        private Registry() {
            workers = new LinkedList<Employee>();
        }

        public static Registry GetInstance() {
            return WorkersRegistry;
        }

        public void AddEmployeer(Employee employeer) {
            if (workers.Contains(employeer)) {
                throw new EmployeeInRegistryException
                    ("Не можна найняти двічі одну й ту саму людину!");
            }
            workers.AddFirst(employeer);
        }

        public void PrintList() {
            Console.WriteLine("Список всіх працівників:");
            foreach (var employeer in workers) {
                Console.WriteLine(employeer);
            }
        }
    }
}